tests/documents/sample_6.xml

```diff
@@ -0,0 +1,3 @@
+<doc>Hello</doc>
+
+<doc>World</doc>
```

Comment:
Actually, this is not a well-formed XML document. See the `document` production: https://www.w3.org/TR/REC-xml/#NT-document. The fact that the library by default does not fail here is actually a bug :) Although I do agree that this might be a useful mode, but it ideally should be controlled by a completely different flag, like "fragment mode" I saw in other XML parsers.

Anyway, I'm okay with merging this right now. I'll do a proper review a bit later. Thanks!

Reply:
Oh oops! I over simplified the example. I was actually dealing with processing instructions, e.g.

```
<?xml version="1.0"?><?xml-stylesheet href="doc.xsl" type="text/xsl" ?>
<!DOCTYPE doc SYSTEM "doc.dtd"><doc>Hello, world!</doc>
```

vs

```
<?xml version="1.0"?>

<?xml-stylesheet href="doc.xsl" type="text/xsl" ?>

<!DOCTYPE doc SYSTEM "doc.dtd">

<doc>Hello, world!</doc>
```

Let me change the sample to reflect that.